### PR TITLE
GH workflow to update grammar targets

### PR DESCRIPTION
**Pull request by svengiegerich**
**Motivation**: Update the grammar automatically (every month or so) via GH workflow.

**Changes**: Add GH workflow to update the grammar regularly.

### CODE REVIEW

File: src/pytsql/grammar/helper_generate_parsers.sh [hunk 1]
```diff
@@ -9,8 +9,8 @@ cd "$( dirname "${BASH_SOURCE[0]}" )"
 antlr4="java -jar $ANTLR4_JAR_FILEPATH"
 
 # Optionally: Download grammar
-# curl -o TSqlParser.g4 https://raw.githubusercontent.com/antlr/grammars-v4/master/sql/tsql/TSqlParser.g4
-# curl -o TSqlLexer.g4 https://raw.githubusercontent.com/antlr/grammars-v4/master/sql/tsql/TSqlLexer.g4
+curl -o TSqlParser.g4 https://raw.githubusercontent.com/antlr/grammars-v4/master/sql/tsql/TSqlParser.g4
```
Comment (ivergara) [inline src/pytsql/grammar/helper_generate_parsers.sh:12]: If this is optional, and previously commented, don't you want to put it into an option of the script?

File: .github/workflows/update_grammar.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                python-version: 3.11
+                miniforge-variant: Mambaforge
+                miniforge-version: 4.11.0-2
+                use-mamba: true
+                environment-file: environment.yml
+                activate-environment: pytsql
+            - name: Download ANTLR
+              working-directory: src/pytsql/grammar
+              run: wget https://repo1.maven.org/maven2/org/antlr/antlr4/4.11.1/antlr4-4.11.1-complete.jar
+            - name: Download and re-generate
```
Comment (ivergara) [inline .github/workflows/update_grammar.yml:29]: Why does this step say "Download"?

**Comment by ivergara** [comment]
We have a very similar action [here](https://github.com/Quantco/ipe.triton-inference-server/blob/main/.github/workflows/triton-version-autoupdate.yml). But don't remember how @JanJaguschQC tested it.

File: .github/workflows/update_grammar.yml [hunk 3]
```diff
@@ -26,8 +26,17 @@ jobs:
             - name: Download ANTLR
               working-directory: src/pytsql/grammar
               run: wget https://repo1.maven.org/maven2/org/antlr/antlr4/4.11.1/antlr4-4.11.1-complete.jar
-            - name: Download and re-generate
+            - name: Download latest grammar definition and re-generate grammar targets
               working-directory: src/pytsql/grammar
-              run: bash helper_generate_parsers.sh antlr4-4.11.1-complete.jar
+              run: bash helper_generate_grammar_targets.sh antlr4-4.11.1-complete.jar -d
             - name: Install package
               run: pip install -e .
```
Comment (ivergara) [inline .github/workflows/update_grammar.yml:33]: Don't think you need the `-e` in here
```suggestion
              run: pip install .
```

File: src/pytsql/grammar/helper_generate_grammar_targets.sh [hunk 4]
```diff
@@ -3,14 +3,18 @@
...
 cd "$( dirname "${BASH_SOURCE[0]}" )"
 
 antlr4="java -jar $ANTLR4_JAR_FILEPATH"
 
-# Optionally: Download grammar
-curl -o TSqlParser.g4 https://raw.githubusercontent.com/antlr/grammars-v4/master/sql/tsql/TSqlParser.g4
-curl -o TSqlLexer.g4 https://raw.githubusercontent.com/antlr/grammars-v4/master/sql/tsql/TSqlLexer.g4
+# Optionally: Download latest grammar definition before generating grammar
+if [[ "$2" == "download" || "$2" == "-d" ]]; then
+  echo "he"
```
Comment (ivergara) [inline src/pytsql/grammar/helper_generate_grammar_targets.sh:14]:
```suggestion
```

**Review by ivergara — COMMENTED** [review]
It looks good to me.

**Comment by ivergara** [comment]
Will let @SimeonStoykovQC have the last word.

**Review by SimeonStoykovQC — APPROVED** [review]
It only gets as awesome as this 🔝 🚀 Great work and overall persistence in modernizing pytsql grammars!

**Comment by SimeonStoykovQC** [comment]
Maybe the environment is not active when the script is run and you have to `conda/mamba activate ...` explicitly?

**Pull request merged**